src/main/java/com/michelin/cio/hudson/plugins/maskpasswords/MaskPasswordsConsoleLogFilter.java

```diff
@@ -69,7 +43,9 @@
  * @author Jason Antman jason@jasonantman.com
  */
 @Extension
-public class MaskPasswordsConsoleLogFilter extends ConsoleLogFilter {
+public class MaskPasswordsConsoleLogFilter extends ConsoleLogFilter  implements Serializable {

```

Comment:
does it really need to be serializable?

Reply:
yes, see https://issues.jenkins-ci.org/browse/JENKINS-45693

Reply:
It is not going to work, since you will be calling `MaskPasswordsConfig.getInstance()` from the agent JVM and this will at best return junk.

To make this actually function, you would need to create a `writeReplace` method which captures `passwords` and `regexes` in a non`transient` instance field, or something similar.